rest_framework-stubs/authentication.pyi

```diff
@@ -1,32 +1,34 @@
 from typing import Any, Optional, Tuple, Type
 
-from django.contrib.auth import authenticate as authenticate
 from django.db.models import Model
 from django.middleware.csrf import CsrfViewMiddleware
-from rest_framework.request import Request
 
-def get_authorization_header(request: Request) -> bytes: ...
+def get_authorization_header(request: Any): ...
 
 class CSRFCheck(CsrfViewMiddleware): ...
 
-class BaseAuthentication(object):
-    def authenticate(self, request: Request) -> Optional[Tuple[Any, Any]]: ...
-    def authenticate_header(self, request: Request) -> Optional[str]: ...
+class BaseAuthentication:
+    def authenticate(self, request: Any) -> None: ...
+    def authenticate_header(self, request: Any) -> None: ...
 
 class BasicAuthentication(BaseAuthentication):
     www_authenticate_realm: str = ...
-    def authenticate_credentials(
-        self, userid: str, password: str, request: Optional[Request] = ...
-    ) -> Tuple[Any, None]: ...
+    def authenticate(self, request: Any): ...
+    def authenticate_credentials(self, userid: Any, password: Any, request: Optional[Any] = ...): ...
+    def authenticate_header(self, request: Any): ...
 
 class SessionAuthentication(BaseAuthentication):
-    def enforce_csrf(self, request: Request) -> None: ...
+    def authenticate(self, request: Any): ...
```

Comment:
There are a lot of `Any`s. Did you use some kind of tooling to sync these two repos?

Reply:
I used Mypy stubgen to generate stubs, I then manually merged the diff.

Reply:
Sadly, there are lots of removed types like in this particular diff 😞 

Do you know any way to fix this? (without manually digging into it?)

Reply:
In this case it was a manual comparison. I will do another pass through all modified files. But some stuff was indeed removed between drf versions.

Should I keep it for backwards compatibility?

Reply:
fixed this file. Going through the entire diff now.